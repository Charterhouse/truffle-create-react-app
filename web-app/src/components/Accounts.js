import React from 'react'
import { Link } from 'react-router-dom'
import withWeb3 from 'web3/withWeb3'
import { Wrapper } from 'components/Wrapper'

// Simple demonstration of the withWeb3 higher-order component
const Accounts = ({ accounts }) =>
  <Wrapper>
    <h1>My Accounts</h1>
    <pre>{JSON.stringify(accounts, null, 4)}</pre>
    <div style={{ marginTop: '20px' }}>
      <Link to='/'>Home</Link>
      <Link to='/daap' style={{ marginLeft: '20px' }}>My DAap</Link>
    </div>
  </Wrapper>

export default withWeb3(Accounts)
