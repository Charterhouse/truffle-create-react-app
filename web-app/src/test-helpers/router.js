import React from 'react'
import { MemoryRouter as Router, Route } from 'react-router-dom'

export const inRouter = (Component, path) => {
  return (
    <Router initialEntries={[ path ]} initialIndex={0}>
      <Route exact path={path} component={Component} />
    </Router>
  )
}

export const renderInRouter = (componentRenderFunction, path) => {
  return (
    <Router initialEntries={[ path ]} initialIndex={0}>
      <Route exact path={path} render={componentRenderFunction} />
    </Router>
  )
}
