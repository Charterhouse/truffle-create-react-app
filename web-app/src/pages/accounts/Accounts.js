import React from 'react'
import { Wrapper } from 'components/wrapper'
import { AppNavigation } from 'components/navigation'

const Accounts = ({ accounts, location }) =>
  <Wrapper>
    <h1>My Accounts</h1>
    <pre>{JSON.stringify(accounts, null, 4)}</pre>
    <AppNavigation location={location} />
  </Wrapper>

export { Accounts }
