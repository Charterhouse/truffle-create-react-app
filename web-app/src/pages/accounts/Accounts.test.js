import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { renderInRouter } from 'test-helpers/router'
import { Accounts } from './Accounts'

const testAccounts = [
  '0x627306090abab3a6e1400e9345bc60c78a8bef57'
]

const renderAccountsInRoute = routeProps =>
  <Accounts accounts={testAccounts} {...routeProps} />

let accountsInRoute

beforeEach(() => {
  accountsInRoute = renderInRouter(
    renderAccountsInRoute,
    '/accounts')
})

it('renders without crashing', async () => {
  const div = document.createElement('div')
  await ReactDOM.render(accountsInRoute, div)
})

it('renders correctly', async () => {
  const accounts = await renderer
    .create(accountsInRoute)
  expect(accounts.toJSON()).toMatchSnapshot()
})
