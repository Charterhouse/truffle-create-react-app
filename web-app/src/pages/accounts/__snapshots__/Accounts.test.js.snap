// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  style={
    Object {
      "padding": "20px",
    }
  }
>
  <h1>
    My Accounts
  </h1>
  <pre>
    [
        "0x627306090abab3a6e1400e9345bc60c78a8bef57"
    ]
  </pre>
  <div
    style={
      Object {
        "marginTop": "20px",
      }
    }
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      href="/dapp"
      onClick={[Function]}
      style={
        Object {
          "marginLeft": "20px",
        }
      }
    >
      My DApp
    </a>
  </div>
</div>
`;
